### PR TITLE
Skip updating web app if runtime not configured in v2 schema version

### PR DESCRIPTION
**Pull request by SummerSun**
#437

### CODE REVIEW

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/WebAppUtils.java [hunk 1]
```diff
@@ -161,6 +163,20 @@ public static RuntimeStack getLinuxRunTimeStack(String imageName) throws MojoExe
...
+            return false;
+        }
+
+        return !StringUtils.isEmpty(runtime.getOs()) || runtime.getJavaVersion() != null ||
+            runtime.getWebContainer() != null || !StringUtils.isEmpty(runtime.getImage()) ||
+            !StringUtils.isEmpty(runtime.getServerId()) || !StringUtils.isEmpty(runtime.getRegistryUrl());
+    }
+
     /**
      * Work Around:
```
Comment (jdneo) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/WebAppUtils.java]: `Workaround`, not `Work Around`

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/WebAppUtils.java [hunk 2]
```diff
@@ -161,6 +163,20 @@ public static RuntimeStack getLinuxRunTimeStack(String imageName) throws MojoExe
...
+    public static boolean isUpdateWebAppNecessary(final String schemaVersion, final RuntimeSetting runtime)
+        throws MojoExecutionException {
+        if (SchemaVersion.V2 != SchemaVersion.fromString(schemaVersion)) {
+            return true;
+        }
+        if (runtime == null) {
+            return false;
+        }
+
+        return !StringUtils.isEmpty(runtime.getOs()) || runtime.getJavaVersion() != null ||
```
Comment (jdneo) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/WebAppUtils.java]: `StringUtils.isNotEmpty()`

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/WebAppUtils.java [hunk 3]
```diff
@@ -161,6 +163,20 @@ public static RuntimeStack getLinuxRunTimeStack(String imageName) throws MojoExe
         throw new MojoExecutionException(IMAGE_NOT_GIVEN);
     }
 
+    public static boolean isUpdateWebAppNecessary(final String schemaVersion, final RuntimeSetting runtime)
```
Comment (jdneo) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/WebAppUtils.java]: What if the runtime setting is just the same as the actual runtime of the Web App

Comment (SummerSun) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/WebAppUtils.java]: Will still do the update. If there are any improvement space, I think it should be handled in the API side.

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/WebAppUtils.java [hunk 4]
```diff
@@ -161,8 +163,22 @@ public static RuntimeStack getLinuxRunTimeStack(String imageName) throws MojoExe
...
+        throws MojoExecutionException {
+        if (SchemaVersion.V2 != SchemaVersion.fromString(schemaVersion)) {
+            return true;
+        }
+        if (runtime == null) {
+            return false;
+        }
+
+        return StringUtils.isNotEmpty(runtime.getOs()) || runtime.getJavaVersion() != null ||
```
Comment (jdneo) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/WebAppUtils.java]: Add a util method: `isEmpty()` in runtime

**Pull request merged**